src/test/java/de/szut/lf8_project/integration/TestUpdateProject.java

```diff
@@ -0,0 +1,262 @@
+        Employee newProjectLead = saveEmployeeInRemoteRepository(createDefaultEmployeeWithout0Id());
+        String jsonUpdateBody = String.format("""
+                {
+                "projectName" : "Neuer Name",
+                "projectDescription" : "Neue Beschreibung",
+                "customerId" : 999,
+                "customerContact" : "Der Neue",
+                "projectLeadId" : %d,
+                "startDate" : "2022-01-13",
+                "plannedEndDate" : "2022-02-14",
+                "actualEndDate" : "2022-03-15"
+                }
+                """, newProjectLead.getId().unbox());
+
+        ResultActions result = mockMvc.perform(
+                put("/api/v1/project/" + project.getProjectId().get().unbox())
+                        .header("Authorization", jwt.jwt())
+                        .content(jsonUpdateBody)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .accept(MediaType.APPLICATION_JSON)
+        );
+        Project projectFromDb = getProjectByIdFromDatabase(project.getProjectId().get());
+
+        result.andExpect(status().isOk());
+        assertThat(projectFromDb.getProjectName().unbox()).isEqualTo("Neuer Name");
+        assertThat(projectFromDb.getProjectDescription().get().unbox()).isEqualTo("Neue Beschreibung");
+        assertThat(projectFromDb.getProjectLead().getProjectLeadId().unbox()).isEqualTo(newProjectLead.getId().unbox());
+        assertThat(projectFromDb.getCustomerContact().unbox()).isEqualTo("Der Neue");
+        assertThat(projectFromDb.getCustomer().getCustomerId().unbox()).isEqualTo(999);
+        assertThat(projectFromDb.getStartDate().get().unbox()).isEqualTo(LocalDate.of(2022,1,13));
+        assertThat(projectFromDb.getPlannedEndDate().get().unbox()).isEqualTo(LocalDate.of(2022,2,14));
+        assertThat(projectFromDb.getActualEndDate().get().unbox()).isEqualTo(LocalDate.of(2022,3,15));
+
+    }
+
+    @ParameterizedTest(name = "für das Feld {0}")
+    @MethodSource
+    @DisplayName("ein Projekt erfolgreich teilweise updaten")
+    void partialUpdate(String fieldToUpdate, Object newValue, String jsonResultPath) throws Exception {
+        Project project = createAndSaveDefaultProjectWithProjectLead();
+        saveProjectInDatabase(project);
+        String jsonUpdateBody = "{\""  + fieldToUpdate + "\": " + (newValue instanceof String ? "\"" + newValue + "\"" : newValue ) + "}";
+
+        ResultActions result = mockMvc.perform(
+                put("/api/v1/project/" + project.getProjectId().get().unbox())
+                        .header("Authorization", jwt.jwt())
+                        .content(jsonUpdateBody)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .accept(MediaType.APPLICATION_JSON)
+        );
+
+        result
+                .andExpect(status().isOk())
+                .andExpect(jsonPath(jsonResultPath, is(newValue)));
```

Comment:
Weiß nicht ob ich so ein Fan davon bin hier nur die Response zu testen und oben nur DB.
Ja, irgendwo SOLLTE es egal sein.
Aber um genau solche (eventuell falschen) Annahmen zu bestätigen schreiben wir ja Tests.
Siehe Schwerks API, wo das Skillset nicht richtig geht.

Bringt mich aber in diesem Falle auch nicht um den Schlaf.
Daher trotzdem approved und als optional anzusehen.

Reply:
ja ich machs mir mal separat auf - trick ist halt: wie macht man das schlau parameterized? falls gute idee gerne her damit!